### ISSUE TITLE
Read csv with custom delimiter

### ISSUE THREAD

**Issue by wieseljonas** [SheetJS/js-xlsx#1087]
I'm trying to read a csv file with '#' as delimiter  I haven't found anyway to specify this in the `read` function.

Is this possible?

**Comment by SheetJSDev** [comment]
The DSV reader follows Excel rules.  For a delimiter other than the standard comma or tab or semicolon, your file should start with `sep=` followed by the separator character.  You can verify this manually in Excel.

This file will have 2 cells:

```
1#2#3
4#5#6
```

This file will have 6 cells:

```
sep=#
1#2#3
4#5#6
```

There currently is no override but it is fairly straightforward to add one and we would accept a PR if you are interested.  To be consistent with the writer, the option should be called `FS`. 
 https://github.com/SheetJS/js-xlsx/blob/master/bits/40_harb.js#L770 is where the delimiter is determined, and the check should apply after the sep override:

```diff
		if(str.slice(0,4) == "sep=" && str.charCodeAt(5) == 10) { sep = str.charAt(4); str = str.slice(6); }
+		else if(o.FS) sep = o.FS;
		else sep = guess_sep(str.slice(0,1024));

```

**Comment by wieseljonas** [comment]
@SheetJSDev Thank you for your quick response this makes sense

**Comment by louisjacksonrees** [comment]
@SheetJSDev  I have just tested this using version 0.15.4 and i get the following cells:
`sep=# | 37623 | 38811.958333333336`

with the following sheet parsing options:
`{ type: "array", sheets: 0, sheetRows: 0 }`

Any help on getting this to work, would be super appreciated

**Comment by louisjacksonrees** [comment]
Upon a deeper look at this issue i can see that the code is checking if there is the char code for a new line present at position 5, this is looking for the char code 10 - whereas my file has a char code of 13 for the new line

**Comment by garrettluu** [comment]
> Upon a deeper look at this issue i can see that the code is checking if there is the char code for a new line present at position 5, this is looking for the char code 10 - whereas my file has a char code of 13 for the new line

@louisjacksonrees this bug was fixed in this PR: #2004
